src/Sarif/AggregatingArtifactsProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class AggregatingArtifactsProvider : IArtifactProvider
```

Comment:
>AggregatingArtifactsProvider [](http://example.com/codeflow?start=17&length=28)

An aggregating artifacts provider is an artifacts provider that itself aggregates scan targets generated by one or more providers. Directly analogous to our aggregating logger.

Note that individual providers own problems such as skipping files which aren't appropriate for scanning (because, for example, they exceed a size threshold) and for reporting those that were skipped.